electrumsv/network.py

```diff
@@ -1027,305 +290,144 @@ async def _main_task(self) -> None:
-        server = SVServer.all_servers.get(existing_key)
-        if server is None:
-            raise KeyError("server does not exist")
-
-        def on_disconnection_completed(*_: Any) -> None:
-            assert server is not None
-            server.remove()
-            if callback is not None:
-                callback()
-
-        server.state.is_disabled = True
-        callback_pending = False
-        if server.add_disconnection_callback(on_disconnection_completed):
-            callback_pending = True
-        server.disconnect()
-        if not callback_pending:
-            on_disconnection_completed()
-
-    def delete_electrumx_server(self, existing_key: SVServerKey,
-            callback: Optional[Callable[[], None]]=None) -> None:
-        app_state.async_.spawn(self.delete_electrumx_server_async, existing_key, callback)
-
-    def _read_config_electrumx(self) -> Tuple[SVServer, Optional[SVProxy]]:
-        # Remove obsolete key
-        config = app_state.config
-        config.set_key('server_blacklist', None)
-        # The way SVServers are populated from config file is confusing. JSON.register() is called
-        # for SVServer and when the config is deserialized, the specially serialised SVServer
-        # entries are instantiated and in doing so they add themselves to the `all_servers` list.
-        logger.info('Read %d electrumx servers from config file', len(SVServer.all_servers))
-        # Add default servers if not present. If we add the ability for users to delete servers
-        # and they want to delete default serves, then this will override that.
-        for host, data in Net.DEFAULT_SERVERS_ELECTRUMX.items():
-            for protocol in 'st':
-                if protocol in data:
-                    SVServer.unique(host, data[protocol], protocol)
-
-        main_server = config.get('server', None)
-        if isinstance(main_server, str):
-            main_server = SVServer.from_string(main_server)
-            config.set_key('server', main_server, True)
-        if not isinstance(main_server, SVServer):
-            logger.info('choosing an SSL server randomly; none in config')
-            # TODO We need a better server selection mechanism where if we choose the secure
-            #   version it falls back to the insecure version if there is one specified.
-            if Net.COIN is BitcoinRegtest:
-                main_server = self._random_server_nowait('t')
-            else:
-                main_server = self._random_server_nowait('s')
-            if not main_server:
-                raise RuntimeError('no servers available')
+    def get_local_tip_hash(self) -> bytes:
+        chain = self.chain()
+        assert chain is not None
+        return cast(bytes, chain.tip.hash)
 
-        proxy = config.get('proxy', None)
-        if isinstance(proxy, str):
-            proxy = SVProxy.from_string(proxy)
+    async def get_esv_server_tip_height(self) -> int:
```

Comment:
There is no such thing as an esv server. Do not refer to esv_server, just use server.

Reply:
This function should have an `_async` suffix.

Reply:
I also do not like anything that checks locally then falls back to remote servers. This is problematic as the caller has no idea if their call will block or not. I suspect the caller should deal with absence of state explicitly.